DuoGamerSDK.m

```diff
@@ -157,36 +168,44 @@ - (void)_accessoryDidDisconnect:(NSNotification *)notification {
 }
 
 -(void)processPacket{
-
-    // valid fixed data, but ignore xor checksum
-    uint8_t * data = (uint8_t*) [readData bytes];
-    if (data[0] != 0x5a) return;
-    if (data[1] != 0xa5) return;
-    if (data[2] != 0x04) return;
-    if (data[9] != 0x00) return;
-
-    DuoGamerState state;
-    uint8_t buttons_1      = data[3];
-    uint8_t buttons_2      = data[4];
-    state.buttonA          = isBitSet(buttons_1, 0);
-    state.buttonB          = isBitSet(buttons_1, 1);
-    state.buttonY          = isBitSet(buttons_1, 2);
-    state.buttonX          = isBitSet(buttons_1, 3);
-    state.analogLeftClick  = isBitSet(buttons_1, 5);
-    state.analogRightClick = isBitSet(buttons_1, 6);
-    state.shoulderRight    = isBitSet(buttons_2,  0);
-    state.shoulderLeft     = isBitSet(buttons_2,  1);
-    state.dpadUp           = isBitSet(buttons_2,  2);
-    state.dpadRight        = isBitSet(buttons_2,  3);
-    state.dpadDown         = isBitSet(buttons_2,  4);
-    state.dpadLeft         = isBitSet(buttons_2,  5);
-    state.analogLeftX      = data[5] - 0x80;
-    state.analogLeftY      = data[6] - 0x80;
-    state.analogRightX     = data[7] - 0x80;
-    state.analogRightY     = data[8] - 0x80;
-
-    if (![delegate respondsToSelector:@selector(handleState:)]) return;
-    [delegate handleState:&state];
+    uint8_t * data = malloc(sizeof(uint8_t));
```

Comment:
This effectively calls malloc(1), reserving a single byte of memory. Then below, 11 bytes are stored at that location creating a buffer overrun. In practice, every malloc'd block will probably have min. 16 bytes, but it's still incorrect.

If you just want an array of size SINGLE_PACKET_SIZE, please just use

uint8_t data[SINGLE_PACKET_SIZE];

you can then also remove the free(data) call later